### ISSUE TITLE
Copy button (clipboard-btn) not working in my instance

### ISSUE THREAD

**Issue by duchenpaul** [go-gitea/gitea#17593]
### Gitea Version

Gitea Version: 1.15.6

### Operating System

Docker, host system: Debian

### How are you running Gitea?

Run Gitea from docker, on amd64 system.

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

No

### Description

I tried to copy the url by clicking the clipboard icon, but it shows `Use ⌘C or Ctrl-C to copy`, instead of `Link has been copied`, there is something wrong somewhere, but not sure.

From the chrome consule there is no error pops up.
From the backend, neither.

### Screenshots

![image](https://user-images.githubusercontent.com/8120099/140863437-3faec24b-b837-4039-a9b8-256b0a211327.png)

**Comment by wxiaoguang** [comment]
If you see `Use ⌘C or Ctrl-C to copy`, it is usually caused by your browser, you browser denied the copy action.

**Comment by duchenpaul** [comment]
ok, how to enable it? I tries this on two devices, no lucks

**Comment by wxiaoguang** [comment]
TBH, I have no idea, the code for 1.16.0+dev-474-g021df2962 is exactly the same as try.gitea.io

The copy code is pretty simple: https://github.com/go-gitea/gitea/blob/main/web_src/js/features/clipboard.js

Can you try the code in your browser?

**Comment by duchenpaul** [comment]
I enabled debug level logging in chrome, saw this error when I click copy button
```
[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
add @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
(anonymous) @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
each @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
each @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
sn @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
on @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
events @ index.js?v=66b87c123c0ab767e55d95c43d170494:1
initialize @ index.js?v=66b87c123c0ab767e55d95c43d170494:1
(anonymous) @ index.js?v=66b87c123c0ab767e55d95c43d170494:1
each @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
each @ index.js?v=66b87c123c0ab767e55d95c43d170494:62
l.fn.popup @ index.js?v=66b87c123c0ab767e55d95c43d170494:1
jh @ index.js?v=66b87c123c0ab767e55d95c43d170494:25
(anonymous) @ index.js?v=66b87c123c0ab767e55d95c43d170494:25
(anonymous) @ index.js?v=66b87c123c0ab767e55d95c43d170494:25
Dc @ index.js?v=66b87c123c0ab767e55d95c43d170494:25
(anonymous) @ index.js?v=66b87c123c0ab767e55d95c43d170494:25
```

**Comment by duchenpaul** [comment]
Will this help? if you can read chinese
https://www.coder.work/article/943540

**Comment by wxiaoguang** [comment]
Not related. If you can not find some methods to reproduce this problem, it would be difficult to continue to debug, since it works on try.gitea.io, the same code.

**Comment by silverwind** [comment]
Might be an issue with the clipboard fallback code that was recently added. I see the instance is running on `http`, so the code will definitely branch to the fallback:

https://github.com/go-gitea/gitea/blob/43bbc5478370cbfe3ab1eed730ea002ccec74708/web_src/js/features/clipboard.js#L25-L46

**Comment by silverwind** [comment]
Can't reproduce. Fallback works fine for me on all major browsers. What version of Chrome is this observed on and does it work in other browsers?

**Comment by tech-meppem** [comment]
This is what I attempted for my fix, because this is what happens on non-https sites.
Are you using your site via Https?
If not, then the copy to clipboard won't work for anything.

or at least, unless you have the fix stated above merged in, which I don't know which version it's actually going to be included in.

**Comment by silverwind** [comment]
OPs version above indicates they have a version with the fix included. Might be worth a try to clear browser cache and try again.

**Comment by duchenpaul** [comment]
Yes, that was my thought as well, I am using http instead of https

**Comment by duchenpaul** [comment]
Using incognico mode not helping as well, and I cannot find anywhere to see the error, including chrome console

**Comment by duchenpaul** [comment]
Correction:
My gitea version is `Gitea Version: 1.15.6`

**Comment by wxiaoguang** [comment]
Please make sure you are using the correct version. The clipboard for non-https was fixed in 1.16

**Comment by duchenpaul** [comment]
Thanks, it works for me when I changed the docker images tag to dev, where the version is `1.16.0+dev-488-g0e189eeca`
